### PR TITLE
Add integration tests for the remote inference feature

### PR DESCRIPTION
**Pull request by ryanbogan**
### Description
Adds integration tests for the remote inference feature
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ylwu-amzn — DISMISSED** [review]

**Comment by ylwu-amzn** [comment]
Take a look at the failed cases? 
```
31 tests completed, 4 failed, 12 skipped

Tests with failures:
 - org.opensearch.ml.rest.RestMLRemoteInferenceIT.testOpenAIModerationsModel
 - org.opensearch.ml.rest.RestMLRemoteInferenceIT.testOpenAIEditsModel
 - org.opensearch.ml.rest.RestMLRemoteInferenceIT.testPredictRemoteModel
 - org.opensearch.ml.rest.RestMLRemoteInferenceIT.testCohereGenerateTextModel
```

**Comment by ryanbogan** [comment]
Those are expected failures until the keys are added to GitHub secrets on the ml-commons repository

**Comment by ylwu-amzn** [comment]
> Those are expected failures until the keys are added to GitHub secrets on the ml-commons repository

Got it , thanks. Who will help add the keys to Github secrets?

**Comment by ryanbogan** [comment]
> > Those are expected failures until the keys are added to GitHub secrets on the ml-commons repository
> 
> Got it , thanks. Who will help add the keys to Github secrets?

It has to be someone with admin permissions on the repository.

**Comment by peterzhuamazon** [comment]
I am adding the key sections to the github workflows after talking to @ryanbogan and @ylwu-amzn offline and make change in their accounts.

**Comment by peterzhuamazon** [comment]
Remove dco.yml as it is replaced by dco apps already.

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java [hunk 1]
```diff
@@ -0,0 +1,675 @@
...
+    }
+
+    private void setEncryptionMasterKey() throws IOException {
+        Response response = TestHelper
+            .makeRequest(
+                client(),
+                "PUT",
+                "_cluster/settings",
+                null,
+                "{\"persistent\":{\"plugins.ml_commons.encryption.master_key\":\"0000000000000011\"}}",
```
Comment (ylwu-amzn) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:665]: We removed this master key in this PR https://github.com/opensearch-project/ml-commons/pull/1075. So this method will fail. Can you rebase and make sure all CI can pass?

**Review by ylwu-amzn — DISMISSED** [review]
LGTM. Thanks Ryan and Peter!

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java [hunk 2]
```diff
@@ -0,0 +1,662 @@
...
+        response = deployRemoteModel(modelId);
+        responseMap = parseResponseToMap(response);
+        taskId = (String) responseMap.get("task_id");
+        waitForTask(taskId, MLTaskState.COMPLETED);
+        response = undeployRemoteModel(modelId);
+        responseMap = parseResponseToMap(response);
+        assertTrue(responseMap.toString().contains("undeployed"));
+    }
+
+    @Ignore
```
Comment (Zhangxunmt) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:184]: The parsing array param bug should be fixed yesterday. Can you check this test again? It should work now.

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java [hunk 3]
```diff
@@ -0,0 +1,662 @@
...
+        responseMap = (Map) responseList.get(0);
+        responseMap = (Map) responseMap.get("dataAsMap");
+        responseList = (List) responseMap.get("results");
+        responseMap = (Map) responseList.get(0);
+        assertTrue((Boolean) responseMap.get("flagged"));
+        responseMap = (Map) responseMap.get("categories");
+        assertTrue((Boolean) responseMap.get("violence"));
+    }
+
+    @Ignore
```
Comment (Zhangxunmt) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:358]: If the problem is still from the post-processing function, can you create a ticket describing this issue? We should fix this problem caught in this test.

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java [hunk 4]
```diff
@@ -0,0 +1,662 @@
...
+        responseMap = (Map) responseList.get(0);
+        responseList = (List) responseMap.get("output");
+        responseMap = (Map) responseList.get(0);
+        responseMap = (Map) responseMap.get("dataAsMap");
+        responseList = (List) responseMap.get("generations");
+        responseMap = (Map) responseList.get(0);
+        assertFalse(((String) responseMap.get("text")).isEmpty());
+    }
+
+    @Ignore
```
Comment (Zhangxunmt) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:478]: The same as above in the chat model test.

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java (hunk 2 above)
Comment (ryanbogan) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:187]: Checking now

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java (hunk 3 above)
Comment (ryanbogan) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:370]: Do you mean a github issue?

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java [hunk 5]
```diff
@@ -0,0 +1,686 @@
...
+        String connectorId = (String) responseMap.get("connector_id");
+        response = TestHelper.makeRequest(client(), "GET", "/_plugins/_ml/connectors/" + connectorId, null, "", null);
+        responseMap = parseResponseToMap(response);
+        assertEquals("OpenAI Connector", (String) responseMap.get("name"));
+        assertEquals("1", (String) responseMap.get("version"));
+        assertEquals("The connector to public OpenAI model service for GPT 3.5", (String) responseMap.get("description"));
+        assertEquals("http", (String) responseMap.get("protocol"));
+    }
+
+    @Ignore
```
Comment (Zhangxunmt) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:85]: This bug is just fixed hours ago. I think you can remove it as well.

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java (hunk 3 above)
Comment (Zhangxunmt) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:370]: Yes.

**Review by Zhangxunmt — APPROVED** [review]

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java (hunk 3 above)
Comment (ryanbogan) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java:369]: Sure I'll do that

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**